src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -926,9 +926,18 @@ private void OutputVirtualSlots(NodeFactory factory, ref ObjectDataBuilder objDa
                     || implType.IsCanonicalSubtype(CanonicalFormKind.Universal)
                     || factory.LazyGenericsPolicy.UsesLazyGenerics(declType)
                     || isInterfaceWithAnEmptySlot)
+                {
                     objData.EmitZeroPointer();
+                }
                 else
-                    objData.EmitPointerReloc(factory.TypeGenericDictionary(declType));
+                {
+                    TypeGenericDictionaryNode dictionaryNode = factory.TypeGenericDictionary(declType);
+                    DictionaryLayoutNode layoutNode = dictionaryNode.GetDictionaryLayout(factory);
+                    if (layoutNode.HasFixedSlots && layoutNode.IsEmpty)
```

Comment:
Does `layoutNode.HasFixedSlots` really needed to be in the condition? I would expect this to work for lazily computed dictionary layout too.

Reply:
We need either this, or condition it on `relocsOnly`. One cannot asks about slots before we have the whole program view.

Reply:
`relocsOnly` would make more sense to me.